src/Servers/Kestrel/Transport.Sockets/src/SocketConnectionListener.cs

```diff
@@ -24,17 +23,20 @@ internal sealed class SocketConnectionListener : IConnectionListener
         private Socket? _listenSocket;
         private int _settingsIndex;
         private readonly SocketTransportOptions _options;
+        private readonly ISocketConnectionContextFactory _contextFactory;
 
         public EndPoint EndPoint { get; private set; }
 
         internal SocketConnectionListener(
             EndPoint endpoint,
             SocketTransportOptions options,
+            ISocketConnectionContextFactory contextFactory,
```

Comment:
The intention behind this API was not to allow people to create their own ConnectionContexts from Sockets, it was to give other Socket-based custom transports an easier time creating ConnectionContexts. We should continue to use SocketConnection directly in SocketConnectionListener.